source/client-side-encryption/client-side-encryption.md

```diff
@@ -1277,6 +1283,11 @@ non-applicable queryType.
 rangeOpts only applies when algorithm is "Range". libmongocrypt returns an error if rangeOpts is set for a
 non-applicable algorithm.
 
+### Custom AWS Credential Provider
+
+When using a `ClientEncryption` object directly and a custom AWS credential provider is specified as an auth mechanism
```

Comment:
It seems slightly odd to pull auth mechanism information from the keyvault client to use on the client encryption for KMS requests.  KMS requests are separate from client auth.  What about users who are using non-AWS auth for their keyvault client (i.e., SCRAM auth) but want to use a custom kms credential provider for KMS requests on their client encryption?

Would it make more sense instead to completely ignore the auth settings on the keyvault client and instead allow ClientEncryption objects to take custom AWS credential providers as an option?

Reply:
Yeah that makes sense, and I think it simplifies this even more. I'll update.

Reply:
I've updated and changed the Node implementation as well. I followed a similar pattern to `KMSProviders` and created a new `CredentialProviders` option that can be provided both to the client encryption options and the auto encryption options. Currently only AWS is supported but I did it in a manner that we could expand to GCP and Azure in the future.